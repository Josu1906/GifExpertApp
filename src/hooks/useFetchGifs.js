import {useState, useEffect} from 'react';
import {getGif} from '../Helpers/getGifs';
export const useFetchGifs = (category) => {
  const [state, setState] = useState({
    data: [],
    loading: true
  });

  useEffect(() => {
    getGif(category)
    .then(imgs =>{
      setTimeout(() => {
        console.log(imgs)
        setState({
          data: imgs,
          loading: false
        })
      }, 3000)
    }
    )
  }, [category]); //useEffect sirve para ejecutar una vez el codigo al renderizar, en las [] le decimos que la vez que cambie lo ejecute
  return state;
}
