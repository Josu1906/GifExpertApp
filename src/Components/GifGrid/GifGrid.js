import React from 'react';
import {GiftGridItem} from '../GiftGridItem/GiftGridItem';
import { useFetchGifs } from '../../hooks/useFetchGifs';
import PropTypes from "prop-types"
// import {getGif} from '../../Helpers/getGifs';

export const GifGrid = ({ category }) => {
  // const [images, setImages] = useState([]);
  //
  const {data, loading} = useFetchGifs(category);
  // useEffect(() => {
  //   getGif(category).then(setImages)
  // }, [category]); //useEffect sirve para ejecutar una vez el codigo al renderizar, en las [] le decimos que la vez que cambie lo ejecute



    return (
      <>
      <h3 className='animate__animated animate__lightSpeedInRight'>Buscaste: {category}</h3>
      {
        loading && <p className='animate__animated animate__lightSpeedInLeft'>Loading...</p>
      }
      <div className='card-grid'>
        {
          data.map(img => (
            <GiftGridItem
            key={img.id}
            {...img} //img={img} Se hace de esta forma pero de esta tambien
            />
          ))
        }
      </div>
      </>
    )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired 
};

// <ol>
//     {
  //       images.map(img => {
    //         // return (
      //         //   <div key={img.id}>
      //         //   <li><b>Id:</b> {img.id}</li>
      //         //   <li><b>Title:</b> {img.title}</li>
      //         //   <li><b>Url:</b> {img.url}</li><br></br>
      //         //   </div>
      //         // )
      //       })
      //     }
      // </ol>
