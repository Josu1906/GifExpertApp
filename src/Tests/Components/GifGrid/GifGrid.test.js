import { shallow } from "Enzyme";
import { GifGrid } from "../../../Components/GifGrid/GifGrid";
import { useFetchGifs } from "../../../hooks/useFetchGifs";
import '@testing-library/jest-dom';

jest.mock('../../../hooks/useFetchGifs')
//Mock actua como una funcion del componente

describe('Pruebas en el componente <GifGrid />', () => {
    const category = 'Josue';
    test('Mostrar la inf en pantalla', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    
    })


    test('Debe mostrar items cuando carga la imagen', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https://www.holasoyJosue.com',
            title: 'Josuee'
        }]; 
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        const wrapper = shallow(<GifGrid category={category}/>)
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
    })
})