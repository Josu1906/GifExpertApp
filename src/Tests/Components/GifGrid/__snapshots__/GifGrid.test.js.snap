// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifGrid /> Mostrar la inf en pantalla 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__lightSpeedInRight"
  >
    Buscaste: 
    Josue
  </h3>
  <p
    className="animate__animated animate__lightSpeedInLeft"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
